### ISSUE TITLE
Allow non-ideal column names

### ISSUE THREAD

**Issue by lazarillo** [tidyverse/dplyr#2719]
This might be similar to [issue 2401](https://github.com/tidyverse/dplyr/issues/2401).

There are older methods which work for very strange or non-ideal column names which will not work within the `tidyverse`.

-----

The following means to select a column in a data frame will fail:

```r
bad_name <- 'This /\r\nOr / That'
my_df <- as_tibble(mtcars)
names(my_df)[2] <- bad_name
my_df %>% select_(bad_name)
```

However, using the "old school" means to select a column will work:

```r
bad_name <- 'This /\r\nOr / That'
my_df[,bad_name]
```

**Comment by hadley** [comment]
You'll need to convert it to a symbol.

**Comment by lazarillo** [comment]
Sorry, I'm not sure what you mean.  I tried some combinations of `eval`, `quote`, or `substitute` to avoid having the `select_` method crash, but everything I tried crashed.

Are you suggesting that I simply alter all of the non-ideal names?  I can do that for REPL-style work, but in this case, I was trying to do something more programmatic and generalized and did not want to choose an overly fragile path.

Thanks!
